addons/account/models/account_move.py

```diff
@@ -4659,8 +4673,12 @@ def default_get(self, default_fields):
             and (self._context.get('journal_id') or self._context.get('default_journal_id')) \
             and self._context.get('default_move_type') in ('out_invoice', 'out_refund', 'in_invoice', 'in_refund', 'out_receipt', 'in_receipt'):
             # Fill missing 'account_id'.
-            journal = self.env['account.journal'].browse(self._context.get('default_journal_id') or self._context['journal_id'])
-            values['account_id'] = journal.default_account_id.id
+            account_id = self.env['account.journal'].browse(self._context.get('default_journal_id') or self._context['journal_id']).default_account_id
+            if self._context.get("default_partner_id"):
+                # Get most used account of the partner as the default.
+                account_id = AccountAccount._get_most_frequent_accounts_by_partner(
+                    self._cr, self.env.company.id, self._context.get("default_partner_id"), self._context.get("default_move_type"), limit=1)[0],
```

Comment:
You can avoid browsing for the journal if you get the account from the partner anyway
```suggestion
            if self._context.get("default_partner_id"):
                # Get the most used account of the partner as the default.
                account_id = AccountAccount._get_most_frequent_accounts_by_partner(
                    self._cr, self.env.company.id, self._context.get("default_partner_id"), self._context.get("default_move_type"), limit=1)[0]
            else:
                account_id = self.env['account.journal'].browse(self._context.get('default_journal_id') or self._context['journal_id']).default_account_id
            
```